.github/workflows/issue-open-check.yml

```
@@ -0,0 +1,48 @@
+name: Issue Open Check
+
+on:
+  issues:
+    types: [opened]
+
+permissions:
+  contents: read
+
+jobs:
+  issue-open-check:
+    permissions:
+      issues: write  # for actions-cool/issues-helper to update issues
+    runs-on: ubuntu-latest
+    steps:
+      - name: check issue title
+        id: check-issue
+        uses: actions/github-script@v3
+        with:
+          script: |
+            const title = context.payload.issue.title;
+            const keywords = ['api', 'treesitter', 'ui', 'lsp', 'doc'];
+            const regex = new RegExp(`\\b(${keywords.join('|')})\\b`, 'i');
+            const match = title.match(regex);
+
+            if (match) {
+              core.setOutput('keyword', match[0]);
+            } else {
+              core.setOutput('keyword', 'bug');
+            }
+
+      - name: add label
+        uses: actions-cool/issues-helper@v3
+        with:
+          actions: 'add-labels'
+          issue-number: ${{ github.event.issue.number }}
+          labels: ${{ steps.check-issue.outputs.keyword }}
```

We really don't need the action, this would achieve the same thing

```suggestion
            const title = context.payload.issue.title;
            const titleSplit = title.split(/\s+/).map(e => e.toLowerCase());
            const keywords = ['api', 'treesitter', 'ui', 'lsp', 'doc'];
            var match = new Set();
            for(const keyword of keywords) {
              if(titleSplit.includes(keyword)) {
                match.add(keyword)
              }
            }
            if(match.size !== 0){
              github.rest.issues.addLabels({
                owner: context.repo.owner,
                repo: context.repo.repo,
                issue_number: context.issue.number,
                labels: Array.from(match)
              })
            }
```